src/pages/resultsView/survival/SurvivalChart.tsx

```diff
@@ -847,45 +872,141 @@ export default class SurvivalChartExtended
 
-        const numberAtRisk = _.groupBy(
-            this.calculateGroupSize(timePoints.map(item => item)),
+    @action.bound
+    calculateLabelWidth(text: number, fontFamily: string, fontSize: number) {
+        const tempElement = document.createElement('div');
+        tempElement.style.position = 'absolute';
+        tempElement.style.opacity = '0';
+        tempElement.style.fontFamily = fontFamily;
+        tempElement.style.fontSize = fontSize.toString();
+        tempElement.textContent = text.toString();
+        document.body.appendChild(tempElement);
+        const labelWidth = tempElement.offsetWidth;
+        document.body.removeChild(tempElement);
+        return labelWidth;
+    }
+
+    @computed get numberOfSamplesAtRisk(): ReactNode[] {
+        const userDefinedTimePoints: number[] = this
+            .timePointsForNumberAtRiskLabels;
+
+        const numberAtRisk: Record<number, RiskPerGroup[]> = _.groupBy(
+            this.calculateGroupSize(userDefinedTimePoints),
             'timePoint'
         );
+        const labelComponents: ReactNode[] = [];
+        let someTimePointsOverlap: boolean = false;
+
+        const checkOverlap = (
+            labelX: number,
+            labelWidth: number,
+            existingLabel: ReactNode
+        ): boolean => {
+            const existingLabelX: number = (existingLabel as any).props.x; // Assuming VictoryLabel has an x attribute
+            const existingLabelWidth: number = this.calculateLabelWidth(
+                (existingLabel as any).props.text,
+                CBIOPORTAL_VICTORY_THEME.legend.style.labels.fontFamily,
+                CBIOPORTAL_VICTORY_THEME.legend.style.labels.fontSize
+            );
 
-        const valueAtAxis = Object.keys(numberAtRisk).map(item =>
-            numberAtRisk[item].map((grp, i) => {
-                return (
-                    <VictoryLabel
-                        text={numberAtRisk[item][i].aliveSamples}
-                        x={
-                            numberAtRisk[item][i].timePoint * this.scaleFactor +
-                            this.styleOpts.padding.left
-                        }
-                        y={
-                            this.styleOptsDefaultProps.height -
-                            this.styleOpts.padding.bottom +
-                            80 +
-                            i * 20
-                        }
-                        style={{
-                            fontFamily:
-                                CBIOPORTAL_VICTORY_THEME.legend.style.labels
-                                    .fontFamily,
-                        }}
-                        textAnchor="middle"
-                    />
-                );
-            })
-        );
+            return !(
+                labelX + labelWidth < existingLabelX ||
+                labelX > existingLabelX + existingLabelWidth
+            );
+        };
 
-        return valueAtAxis;
+        const addLabelsForTimePoint = (
```

Comment:
It would be nice if we can add a comment somewhere in this function to describe how we decide which labels to hide.

Also, we had a similar issue with the lollipop plot labels. I was wondering if we could generalize that logic and use it here as well. It would probably require some refactoring though. So maybe okay to have two separate implementations for now.

https://github.com/cBioPortal/cbioportal-frontend/blob/3e47dd12cb0c1ba13bcff802414e621afb4f2dd9/packages/react-mutation-mapper/src/component/lollipopPlot/LollipopPlotNoTooltip.tsx#L293

Reply:
Added the explanation on when labels are shown/hidden. Will have to compare to logic of `caculateTicks`